pkg/compactor/bucket_compactor.go

```diff
@@ -407,6 +419,13 @@ func (c *BucketCompactor) runCompactionJob(ctx context.Context, job *Job) (shoul
 		return false, nil, err
 	}
 
+	// Check that the minTime and maxTime from the input blocks
+	// are found at least once in the compacted output blocks to ensure
+	// there was no error during concurrent loading of input blocks.
+	if !inputBlocksMinTimeFound.Load() || !inputBlocksMaxTimeFound.Load() {
```

Comment:
The fact we've found a block matching min time and a block matching max time, doesn't mean that there could be blocks outside this range. Also in the following log line we're not giving much information about what was expected, and what we got.

I suggest the following improvement. For each compacted block (which is a block that we're going to upload) ensure that its min and max time are within the min/max time across all input blocks. Then, as you're already doing now, ensure that we have at least 1 compacted block matching the source blocks min time and 1 compacted  block matching the source blocks max time. Whenever any of these validation fail, return an error including which block doesn't match which check, what's the actual value and what's the expected one.

Given the validation logic will be slightly longer than the current one, you can also move all the validation logic into a dedicate function which you can call before the call to `convertCompactionResultToForEachJobs()`.

Reply:
I suggested @fayzal-g the simple approach implemented in this PR. I agree your suggestion is better.